File: output/xCodeEval_8000/C++/C/2af7e700e817e767244391f8ba5f3862.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

typedef struct Vert {
    int hoge[5];
} Vert;

int sign(Vert a, Vert b) {
    int hoge = 0;
    for (int q = 0; q < 5; q++) {
        hoge += a.hoge[q] * b.hoge[q];
    }
    return hoge <= 0;
}

void solve() {
    int n;
    scanf("%d", &n);
    Vert inputs[n];
    for (int i = 0; i < n; i++) {
        for (int q = 0; q < 5; q++) {
            scanf("%d", &inputs[i].hoge[q]);
        }
    }
    int ans[n];
    int ans_count = 0;
    for (int i = 0; i < n; i++) {
        int ok = 1;
        for (int q = 0; q < n; q++) {
            if (i == q) continue;
            for (int j = q + 1; j < n; j++) {
                if (i == j) continue;
                Vert A, B;
                for (int t = 0; t < 5; t++) {
                    A.hoge[t] = inputs[q].hoge[t] - inputs[i].hoge[t];
                    B.hoge[t] = inputs[j].hoge[t] - inputs[i].hoge[t];
                }
                if (sign(A, B) == 0) {
                    ok = 0;
                    break;
                }
            }
            if (ok == 0) break;
        }
        if (ok == 1) {
            ans[ans_count] = i;
            ans_count++;
        }
    }
    printf("%d\n", ans_count);
    for (int i = 0; i < ans_count; i++) {
        printf("%d ", ans[i] + 1);
    }
    printf("\n");
}

int main() {
    solve();
    return 0;
}

File: output/GPT4o-Mini/xCodeEval_4000/C++/Python/2af7e700e817e767244391f8ba5f3862.py
import sys
import math
from collections import deque, defaultdict
from itertools import combinations

def init():
    sys.stdin = open(0)
    sys.stdout = open(1)
    print("{:.20f}".format(0))

class Vert:
    def __init__(self, values):
        self.hoge = values

def sign(a, b):
    return sum(a.hoge[q] * b.hoge[q] for q in range(5)) <= 0

def solve():
    n = int(input())
    inputs = []
    for _ in range(n):
        a = list(map(int, input().split()))
        inputs.append(Vert(a))
    
    ans = []
    for i in range(n):
        ok = 1
        for q in range(n):
            if i == q:
                continue
            for j in range(q + 1, n):
                if i == j:
                    continue
                A = Vert([inputs[q].hoge[t] - inputs[i].hoge[t] for t in range(5)])
                B = Vert([inputs[j].hoge[t] - inputs[i].hoge[t] for t in range(5)])
                if sign(A, B) == 0:
                    ok = 0
                    break
            if ok == 0:
                break
        if ok == 1:
            ans.append(i)
    
    print(len(ans))
    print(" ".join(str(x + 1) for x in ans))

if __name__ == "__main__":
    init()
    solve()
